### DISCUSSION TITLE
zip command with unknown -i option Why?

### DISCUSSION THREAD

**Discussion by wadakatu** [symfony/symfony#45877] [Q&A] [answered]
Hello.
I use Laravel 9 which contains Symfony 6.
I cannot create zip file with wildcard when I write a code something like below.

```php
new Process(['zip', '-jrq', "-P", 'pasword', '/path/to/test.zip', 'path/to/test*.csv']);
```

After mustRun() execute, Error occurs.
error output is something like this.

```
zip error: Notiong to do ! (try: zip -jrq -P password /path/to/test.zip . -i  /path/to/test*.csv)
```

In this error output, there is a unknown option '-i' which I didn't add in a code.
If I use this zip command without -i option on console, it works as I expected and get zip file.

Does anyone know how to get rid of this unknown -i option?

Thank you

**Comment by stof** [comment, marked as answer]
Well, which command are you trying in your terminal ?

- passing the wildcard to the zip command: `zip -jrq -P password /path/to/test.zip '/path/to/test*.csv'`
- letting the shell handle the wildcard and passing multiple files to the zup command `zip -jrq -P password /path/to/test.zip /path/to/test*.csv`

If the test you made is the second one, the command you tried is *not* the one you ran with the Process component (because the process component runs the command directly, so there is no shell processing of the command).
And the zip command suggest you to use the `-i` option in its error message as the way to do a blog-based filtering of included file in the zip command itself.

It is not an unknown option. It is the suggestion solution.

**Reply by wadakatu** [reply]
@stof 
Thank you for your reply.

As you said, I tried the second one.
I didn't know that there is a suggestion in error output.
Then, I finally understand the reason why there is a word 'try:' in the error output.

However, when I use Laravel 5 which contains Symfony 3.4, The code worked without -i option.

Do you know the reason why the -i option is needed now?

**Reply by stof** [reply]
Well, in Symfony 3.4, passing the command as a single string was passing it to the shell for pre-processing (but then needing extra escaping when the shell processing was not expected).
When passing the arguments as an array, the command is executed without wrapping it in a shell. And so you need to use the feature of the `zip` command supporting glob filtering (or you need to resolve the glob separately before building the command, doing the equivalent of what the shell globbing does)

**Reply by wadakatu** [reply]
I am not familiar with pre-processing.
What is it for?

By the way, my command works with process class thanks to your advice.
my latest code is like this.

```php
new Process(['zip', '-jrq', "-P", 'password', '/path/to/test.zip', '.' , '-i' ,'path/to/test*.csv'], 'path/to/');
```

You saved my day!
Thank you very much.
